### DISCUSSION TITLE
MacOS displays GTK Adwaita style window

### DISCUSSION THREAD

**Discussion by jssilva25** [zim-desktop-wiki/zim-desktop-wiki#1581] [General]
Hello,

It's an honor to be the first to discuss on this place; it's a great initiative. Thank you for a very, very, useful and reliable tool.

I am on MacOS Catalina and I just upgraded to 0.74.0 to discover that the window style changed, from the MacOS style I had before, to the Gnome Adwaita style.

If that matters, I'm executing from source in the /Applications/Zim.app/Contents/Resources/zim-0.74.0 folder with python38 and, if I execute from the zim-0.73.5 folder, the style goes back to MacOS.

```
Applications $ cat Zim.app/Contents/Resources/script
#!/bin/sh
cd /Applications
LC_ALL=pt_PT.UTF-8 /opt/local/bin/python3 Zim.app/Contents/Resources/zim-0.74.0/zim.py
#LC_ALL=pt_PT.UTF-8 /opt/local/bin/python3 Zim.app/Contents/Resources/zim-0.73.5/zim.py
```

I must say I privilege functionality against aesthetics but, if I can have both, it's better.

I have in place:
```
$ port installed | grep gtk
  gtk-doc @1.32_1+python38 (active)
  gtk-osx-application-common-gtk2 @2.0.8_0
  gtk-osx-application-common-gtk3 @2.0.8_0 (active)
  gtk-osx-application-gtk2 @2.0.8_0 (active)
  gtk-osx-application-gtk3 @2.0.8_0 (active)
  gtk2 @2.24.32_0+quartz (active)
  gtk3 @3.24.23_0+quartz (active)
  gtkmm3 @3.24.2_0+quartz (active)
```

Am I missing something?
Thank you.

**Comment by jssilva25** [comment]
Forgot to show ./test.py result, but it's the same as it is with 0.73.5, no gi module, but works:
```
zim-0.74.0 $ ./test.py
WARNING: foo
Traceback (most recent call last):
  File "./test.py", line 207, in <module>
    main()
  File "./test.py", line 98, in main
    import tests
  File "/Applications/Zim.app/Contents/Resources/zim-0.74.0/tests/__init__.py", line 170, in <module>
    from zim.newfs import LocalFolder
  File "/Applications/Zim.app/Contents/Resources/zim-0.74.0/zim/newfs/__init__.py", line 21, in <module>
    from .helpers import *
  File "/Applications/Zim.app/Contents/Resources/zim-0.74.0/zim/newfs/helpers.py", line 20, in <module>
    from zim.signals import SignalEmitter, SIGNAL_NORMAL
  File "/Applications/Zim.app/Contents/Resources/zim-0.74.0/zim/signals.py", line 7, in <module>
    from gi.repository import GObject
ModuleNotFoundError: No module named 'gi'

```

**Comment by jaap-karssenberg** [comment]
The Gtk style is not set in zim itself. So are both folders using the same Gtk installation and same Gtk configuration files? If not, you can probably create the Gtk configuration to switch the theme.

Hope this helps,

Jaap

**Comment by jssilva25** [comment]
Thank you for replying.

As I said before, both folders live side-by-side in /Applications/Zim.app/Contents/Resources/

Switching from one to the other is simply made by commenting one and uncommenting the other in /Applications/Zim.app/Contents/Resources/script:
```
#!/bin/sh
cd /Applications
LC_ALL=pt_PT.UTF-8 /opt/local/bin/python3 Zim.app/Contents/Resources/zim-0.74.0/zim.py
#LC_ALL=pt_PT.UTF-8 /opt/local/bin/python3 Zim.app/Contents/Resources/zim-0.73.5/zim.py

```
So it's obviously the same installation and configuration, unless there's something else inside the source folders.

**Comment by jaap-karssenberg** [comment]
That is truly weird. You can search the folder for a "settings.ini" or grep
for "Adwaita" but both are not present in the source repository.

Only other thing I can think of is that the use of the Gtk3 headerbar has
some weird side effect on MacOs - but I know to little of the rendering to
know if that is even possible.

**Comment by jssilva25** [comment]
Yes, weird. Neither do I know about rendering.

So, I suppose will just have to leave it until someone else figures what's wrong.

**Reply by dehesselle** [reply]
This has been resolved in the newest version, you can disable `Show controls in the window decoration` in the toolbar plugin settings.
<img width="345" alt="toolbar" src="https://user-images.githubusercontent.com/11375494/139561555-23aafa97-f660-4177-a542-f24724625e4a.png">

**Reply by rasisin** [reply]
By the way @dehesselle maybe you could start a Discussion topic announcing your unofficial MacOS installer? That would give users a convenient place to give feedback while testing it out.

**Reply by dehesselle** [reply]
@rasisin There's already an issue for that: https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/990
